### DISCUSSION TITLE
Does lancedb have the concept of sharding?

### DISCUSSION THREAD

**Discussion by verawu** [lancedb/lancedb#1453] [Q&A]
I'm required to do a comparison between lancedb and ES, since both can do FTS & vector search.  
One major difference is ES can scale horizontally since its basically a share-nothing architecture, lancedb-oss is more of an embedded db that needs to share resources with application's instance. 

I wonder how lancedb scale on on large scale of data, say PB scale and above. Is Lambda Cloud or Lambda Enterprise architecturally different from lance-oss? Does lancedb have the concept of sharding similar to ES?

**Comment by wjones127** [comment]
LanceDB has a separate of storage and compute. Compute can be horizontally scaled and doesn't have to share anything. Storage needs to be shared, but can be a distributed system like S3 or NFS.
